web/src/portico/landing-page.ts

```diff
@@ -8,71 +8,88 @@ import iphone_image from "../../images/app-screenshots/zulip-iphone-rough.png";
 import {page_params} from "../page_params";
 
 import {detect_user_os} from "./tabbed-instructions";
+import type {UserOS} from "./tabbed-instructions";
 import render_tabs from "./team";
 
-export function path_parts() {
+export function path_parts(): string[] {
     return window.location.pathname.split("/").filter((chunk) => chunk !== "");
 }
 
-const apps_events = function () {
-    const info = {
+type ZulipOsSpecificInfo = Record<
```

Comment:
Need some information on how to best write this. If we use strict types and fields for each specific OS, then we'll have type errors in `update_page` function below. We'll have to change the implemention of `update_page` function to check for type of `OS` version variable represents and only then make dom manipulations to elements specific to that `OS` version.